File: Watcher/src/WeatherData.java
import java.util.ArrayList;

import Interfaces.*;

public class WeatherData implements ISubject{

	private ArrayList<IObserver> observers;
	
	private float temperature;
	private float humidity;
	private float pressure;
	
	public WeatherData() {
		observers = new ArrayList<IObserver>();
	}
	
	@Override
	public void registerObserver(IObserver observer) {
		if (observer != null) {
			observers.add(observer);
		}
		
	}

	@Override
	public void removeObserver(IObserver observer) {
		
		if (observers.contains(observer)) {
			observers.remove(observer);
		}		
	}

	@Override
	public void notifyObservers() {
		for (int i = 0; i < observers.size(); i++) {
			//IObserver observer = (IObserver)observers.get(i);
			//observer.update(temperature, pressure, pressure);
			observers.get(i).update(temperature, humidity, pressure);
		}
		
	}
	
	public void measurementsChanged() {
		notifyObservers();
	}
	
	public void setMeasurements(float temperature, float humidity, float pressure) {
		this.temperature = temperature;
		this.humidity = humidity;
		this.pressure = pressure;
		measurementsChanged();
	}

}

File: c#/Watcher/Models/WeatherData.cs
using System;
using System.Collections.Generic;
using System.Text;
using Watcher.Interfaces;

namespace Watcher.Models
{
    public class WeatherData : IWeatherObservable
    {
        private float Temperature { get; set; }
        private float Humidity { get; set; }
        private float Pressure { get; set; }

        private List<IWeatherObserver> Observers { get; set; }

        public WeatherData()
        {
            Observers = new List<IWeatherObserver>();
        }

        public void AddObserver(IWeatherObserver obs)
        {
            Observers.Add(obs);
        }

        public void NotifyObservers()
        {
            foreach (var wo in Observers)
            {
                wo.Update(Temperature, Humidity, Pressure);
            }
        }

        public void RemoveObserver(IWeatherObserver obs)
        {
            Observers.Remove(obs);
        }

        public void SetMeasurements(float temperature, float humidity, float pressure)
        {
            Temperature = temperature;
            Humidity = humidity;
            Pressure = pressure;
            NotifyObservers();
        }
    }
}
